addons/mail/static/src/models/partner/partner.js

```diff
@@ -168,6 +256,81 @@ function factory(dependencies) {
             return this.user.getChat();
         }
 
+        /**
+         * Returns the text that identifies this partner in a mention.
+         *
+         * @returns {string}
+         */
+        getMentionText() {
+            return this.name;
+        }
```

Comment:
1st step for actual model of mention/suggestion, until @phenix-factory finishes this work ☺️